frontend/src/metabase/nav/containers/MainNavbar/BookmarkList/BookmarkList.tsx

```diff
@@ -28,56 +35,102 @@ interface CollectionSidebarBookmarksProps {
   onDeleteBookmark: (bookmark: Bookmark) => void;
 }
 
+interface BookmarkItemProps {
+  bookmark: Bookmark;
+  index: number;
+  isSorting: boolean;
+  selectedItem?: SelectedEntityItem;
+  onSelect: () => void;
+  onDeleteBookmark: (bookmark: Bookmark) => void;
+}
+
 const BOOKMARKS_INITIALLY_VISIBLE =
   localStorage.getItem("shouldDisplayBookmarks") !== "false";
 
+const BookmarkItem = ({
+  bookmark,
+  index,
+  isSorting,
+  selectedItem,
+  onSelect,
+  onDeleteBookmark,
+}: BookmarkItemProps) => {
+  const { id, item_id, name, type } = bookmark;
+  const isSelected =
+    selectedItem && selectedItem.type === type && selectedItem.id === item_id;
+  const url = Urls.bookmark(bookmark);
+  const icon = Bookmarks.objectSelectors.getIcon(bookmark);
+  const onRemove = () => onDeleteBookmark(bookmark);
+
+  const isIrregularCollection =
+    bookmark.type === "collection" &&
+    !PLUGIN_COLLECTIONS.isRegularCollection(bookmark);
+
+  return (
+    <SortableBookmarkItem index={index} key={bookmark.id}>
+      <SidebarBookmarkItem
+        key={`bookmark-${id}`}
+        url={url}
+        icon={icon}
+        isSelected={isSelected}
+        isSorting={isSorting}
+        hasDefaultIconStyle={!isIrregularCollection}
+        onClick={onSelect}
+        left={<DragIcon name="grabber2" size={12} />}
+        right={
+          <button onClick={onRemove}>
+            <Tooltip tooltip={t`Remove bookmark`} placement="bottom">
+              <Icon name="bookmark" />
+            </Tooltip>
+          </button>
+        }
+      >
+        {name}
+      </SidebarBookmarkItem>
+    </SortableBookmarkItem>
+  );
+};
+
 const BookmarkList = ({
   bookmarks,
   selectedItem,
   onSelect,
   onDeleteBookmark,
 }: CollectionSidebarBookmarksProps) => {
+  const [orderedBookmarks, setOrderedBookmarks] = useState(bookmarks);
+  const [isSorting, setIsSorting] = useState(false);
+
+  useEffect(() => {
+    setOrderedBookmarks(bookmarks);
+  }, [bookmarks]);
+
   const onToggleBookmarks = useCallback(isVisible => {
     localStorage.setItem("shouldDisplayBookmarks", String(isVisible));
```

Comment:
It would be great to move it to user-settings on the backend

Reply:
Yup, haven't forgotten that, thanks for the reminder.